pkg/cloud-init/cloud-init.go

```diff
@@ -67,7 +68,30 @@ func defaultIsoFunc(isoOutFile string, inFiles []string) error {
 	args = append(args, inFiles...)
 
 	cmd := exec.Command("genisoimage", args...)
-	return cmd.Run()
+
+	err := cmd.Start()
+	if err != nil {
+		logging.DefaultLogger().V(2).Error().Reason(err).Msg(fmt.Sprintf("genisoimage cmd failed to start while generating iso file %s", isoOutFile))
+		return err
+	}
+
+	done := make(chan error)
+	go func() { done <- cmd.Wait() }()
+
+	timeout := time.After(10 * time.Second)
+
+	select {
+	case <-timeout:
+		logging.DefaultLogger().V(2).Error().Msg(fmt.Sprintf("Timed out generating cloud-init iso at path %s", isoOutFile))
+		cmd.Process.Kill()
```

Comment:
I assume this triggers and error on the done channel, but select won't pick it up since it already chose the timeout pathway. perhaps this case should just return an error after killing the subprocess?

Reply:
oh nice, good catch! I'll fix this.